src/DictVectors/LoadBalancer.jl

```diff
@@ -0,0 +1,781 @@
+using Rimu
+using Rimu.DictVectors: Communicator, PDWorkingMemory, localpart
+using Rimu.DictVectors:
+    remote_segments, local_segments,
+    first_column, dict_add!, fastrange_hash
+
+using MPI
+using Statistics
+using Random
+import Rimu.DictVectors: mpi_rank, mpi_size, mpi_comm, synchronize_remote!, copy_to_local!, remote_segments,local_segments, first_column, dict_add!, fastrange_hash
```

Comment:
I'm wondering whether `LoadBalancer` should be its own module. Maybe it is OK like this where it is part of the `DictVectors` module. However, if it stays like  this, then all the using and import statements should be moved to `DictVectors.jl` (and some of the statements won't be needed).